internal/provider/packer/resources/bucket/resource_packer_bucket.go

```diff
@@ -0,0 +1,244 @@
+		)
+
+		return
+	}
+
+	r.client = client
+}
+
+func (r *resourcePackerBucket) ModifyPlan(ctx context.Context, req resource.ModifyPlanRequest, resp *resource.ModifyPlanResponse) {
+	modifiers.ModifyPlanForDefaultProjectChange(ctx, r.client.Config.ProjectID, req.State, req.Config, req.Plan, resp)
+}
+
+type bucket struct {
+	ProjectID      types.String `tfsdk:"project_id"`
+	OrganizationID types.String `tfsdk:"organization_id"`
+	Name           types.String `tfsdk:"name"`
+	ResourceName   types.String `tfsdk:"resource_name"`
+	CreatedAt      types.String `tfsdk:"created_at"`
+}
+
+func (r *resourcePackerBucket) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan bucket
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	projectID := r.client.Config.ProjectID
+	if !plan.ProjectID.IsUnknown() {
+		projectID = plan.ProjectID.ValueString()
+	}
+
+	orgID := r.client.Config.OrganizationID
+	loc := &sharedmodels.HashicorpCloudLocationLocation{
+		OrganizationID: orgID,
+		ProjectID:      projectID,
+	}
+	name := plan.Name.ValueString()
+	res, err := packerv2.CreateBucket(ctx, r.client, loc, name)
+	if err != nil {
+		resp.Diagnostics.AddError("Error creating bucket", err.Error())
+		return
+	}
+
+	plan.ResourceName = types.StringValue(res.ResourceName)
+	plan.ProjectID = types.StringValue(res.Location.ProjectID)
+	plan.OrganizationID = types.StringValue(res.Location.OrganizationID)
+	plan.CreatedAt = types.StringValue(res.CreatedAt.String())
+
+	// Save data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
+}
+
+func (r *resourcePackerBucket) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state bucket
+
+	// Read Terraform prior state into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resourceName := state.ResourceName.ValueString()
+	resourceNameRegex := `packer\/project\/.*\/bucket\/.*`
+	// packer/project/{project_id}/bucket/{bucket_name}
+
+	resourceParts := strings.SplitN(resourceName, "/", 5)
+	resourceNameValid, err := regexp.MatchString(resourceNameRegex, resourceName)
+	if err != nil {
+		resp.Diagnostics.AddError("Unexpected provider error", "Failed to parse regular expression, this is an internal error, please report this issue to the provider developers")
+		return
+	}
+	if !resourceNameValid || len(resourceParts) != 5 {
+		resp.Diagnostics.AddError("Invalid resource name", "Resource name expected to match packer/project/{project_id}/bucket/{bucket_name}")
+		return
+	}
+	params := packerservice.NewPackerServiceGetBucketParams()
+	params.SetLocationOrganizationID(state.OrganizationID.ValueString())
+
+	bucketName := resourceParts[4]
+
+	params.SetBucketName(bucketName)
```

Comment:
Why you didn't use the Name state instead? 🤔 
```suggestion
	params.SetBucketName(state.Name.ValueString())
```

Reply:
Because State isn't always set, Read is also called on import and in that case we won't have anything but the resource name